src/pytorch_lightning/trainer/setup.py

```diff
@@ -135,6 +136,7 @@ def init_profiler(trainer: Any, profiler: Optional[Union[Profiler, str]]) -> Non
             "advanced": AdvancedProfiler,
             "pytorch": PyTorchProfiler,
             "xla": XLAProfiler,
+            "hpu": HPUProfiler,
```

Comment:
Since `HPUProfiler` subclasses `PyTorchProfiler` (unlike `XLAProfiler` subclasses `Profiler`), would it make sense to have `Trainer(profiler="pytorch", accelerator="hpu")` pick `HPUProfiler` under the hood so that the code stays accelerator-agnostic? (in a seperate PR)

Reply:
Nice suggestion. This would become applicable for all strategies that are inherited too. If we are able to upstream our changes to PyTorch, then there can be seamless switch in the trainer class as you have pointed.